### ISSUE TITLE
Travis CI examples deployment isn't working

### ISSUE THREAD

**Issue by sasha240100** [WhitestormJS/whitestorm.js#317]
Travis CI examples deployment isn't working.

###### Version:
 - [x] v2.x.x
 - [ ] v1.x.x

###### Issue type:
 - [x] Bug
 - [ ] Proposal/Enhancement
 - [ ] Question
 - [ ] Discussion

------

<details> 
<summary> <b>Tested on: </b> </summary>

###### Desktop
   - [ ] Chrome
   - [ ] Chrome Canary
   - [ ] Chrome dev-channel
   - [ ] Firefox
   - [ ] Opera
   - [ ] Microsoft IE
   - [ ] Microsoft Edge

###### Android
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Opera

###### IOS
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Opera

</details>

**Comment by hirako2000** [comment]
has it ever worked automatically? I don't see the deploy step for *examples*.
I see two *surge* steps. Does not look like they are for pushing examples.

```javascript
before_deploy:
  - gulp examples:build
  - gulp docs --all
...
deploy:
  provider: surge
  project: ./
  domain: whs-dev.surge.sh
  skip_cleanup: true
  on: dev
deploy:
  provider: surge
  project: ./docs/public/
  domain: whsjs.io
  skip_cleanup: true
  on:
    branch: master
    tags: true
```

**Comment by hirako2000** [comment]
Also, I propose we do this as part of this issue:

- Add *another* surge deploy step for examples, to build example that match latest released version of `whs`

```javascript
deploy:
  provider: surge
  project: ./examples
  domain: whs-master.surge.sh
  skip_cleanup: true
  on: master
```

Otherwise some newer examples not supported yet by released version might be tried and would not work on the pulled dependency by users. would be confusing.

**Comment by sasha240100** [comment]
@hirako2000 I completely agree on that why. I still can't figure out why dev examples are not pushing to surge. (something happened, while nothing was touched)

**Comment by hirako2000** [comment]
this is a change, it was touched. maybe we can't have multiple deploy steps with same provider: [diff](https://github.com/WhitestormJS/whs.js/commit/77cd3458f4f42a204b507e2bd65b945c4a42485b#diff-354f30a63fb0907d4ad57269548329e3).

I suggest we do this below, as documented [here](https://surge.sh/help/integrating-with-travis-ci), and using Travis [env variables](https://docs.travis-ci.com/user/environment-variables/) for conditional branch deployment.

```yaml
after_success:
  - test $TRAVIS_BRANCH = "dev" && surge --project ./ --domain whs-dev.surge.sh
  - test $TRAVIS_BRANCH = "master" && surge --project ./ --domain whs-master.surge.sh
  - test $TRAVIS_BRANCH = "master" && surge --project ./docs/public/ --domain whsjs.io 
```
